### PR TITLE
Code refactor and minor optimization

### PR DESCRIPTION
**Pull request by GermanAizek**
This Pull Request adds a few things:

- Modernize sections code
- Fewer object allocations
- Cleaning unused variables

### CODE REVIEW

File: src/common/fiber.cpp [hunk 1]
```diff
@@ -135,7 +135,7 @@ void Fiber::YieldTo(std::weak_ptr<Fiber> weak_from, Fiber& to) {
 }
 
 std::shared_ptr<Fiber> Fiber::ThreadToFiber() {
-    std::shared_ptr<Fiber> fiber = std::shared_ptr<Fiber>{new Fiber()};
+    auto fiber = std::make_shared<Fiber>(Fiber());
```
Comment (merryhime) [inline src/common/fiber.cpp:138]:
```suggestion
    auto fiber = std::make_shared<Fiber>();
```

File: src/core/frontend/applets/error.cpp [hunk 2]
```diff
@@ -8,21 +8,21 @@ namespace Core::Frontend {
 
 ErrorApplet::~ErrorApplet() = default;
 
-void DefaultErrorApplet::ShowError(ResultCode error, std::function<void()> finished) const {
+void DefaultErrorApplet::ShowError(const ResultCode& error, std::function<void()> finished) const {
```
Comment (merryhime) [inline src/core/frontend/applets/error.cpp:11]: Transforming `ResultCode` -> `const ResultCode&` is a pessimization, ResultCode is a u32.

File: src/core/frontend/applets/error.cpp [hunk 3]
```diff
@@ -8,21 +8,21 @@ namespace Core::Frontend {
...
     LOG_CRITICAL(
         Service_Fatal,
         "Application requested error display: {:04X}-{:04X} (raw={:08X}) with timestamp={:016X}",
         error.module.Value(), error.description.Value(), error.raw, time.count());
 }
 
-void DefaultErrorApplet::ShowCustomErrorText(ResultCode error, std::string main_text,
-                                             std::string detail_text,
+void DefaultErrorApplet::ShowCustomErrorText(const ResultCode& error, const std::string& main_text,
+                                             const std::string& detail_text,
```
Comment (merryhime) [inline src/core/frontend/applets/error.cpp:25]: `std::string_view` may be preferable to `const std::string&`.

File: src/core/hle/service/ssl/ssl.cpp [hunk 4]
```diff
@@ -101,7 +101,6 @@ class ISslContext final : public ServiceFramework<ISslContext> {
     void ImportServerPki(Kernel::HLERequestContext& ctx) {
         IPC::RequestParser rp{ctx};
         const auto certificate_format = rp.PopEnum<CertificateFormat>();
-        const auto pkcs_12_certificates = ctx.ReadBuffer(0);
```
Comment (merryhime) [inline src/core/hle/service/ssl/ssl.cpp:104]: One would prefer if one didn't do this.

**Review by merryhime — CHANGES_REQUESTED** [review]
While there are some modifications here which are acceptable, there does not appear to have been a human in the loop selecting which ones are valid and appropriate.

I have not done a complete review.

**Review by v1993 — CHANGES_REQUESTED** [review]
Some of the changes concern replacing passing by value by passing by const reference - in most cases, it's either insignificant or (for smaller structures/classes) a pessimization.

File: src/video_core/rasterizer_interface.h [hunk 5]
```diff
@@ -136,6 +136,11 @@ class RasterizerInterface {
 
     /// Initialize disk cached resources for the game being emulated
     virtual void LoadDiskResources(u64 title_id, std::stop_token stop_loading,
-                                   const DiskResourceLoadCallback& callback) {}
+                                   const DiskResourceLoadCallback& callback) {
+        // Loading was stopped
+        if (stop_loading.stop_requested()) {
+            return;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (GermanAizek) [reply src/video_core/rasterizer_interface.h:144]: @v1993,
References:

https://github.com/yuzu-emu/yuzu/blob/9c701774562ea490296b9cbea3dbd8c096bc4483/src/core/cpu_manager.cpp#L348-L353

https://github.com/yuzu-emu/yuzu/blob/9c701774562ea490296b9cbea3dbd8c096bc4483/src/core/hle/kernel/service_thread.cpp#L46-L50

https://github.com/yuzu-emu/yuzu/blob/9c701774562ea490296b9cbea3dbd8c096bc4483/src/core/hle/kernel/service_thread.cpp#L60-L64

[the remark this answers is not in the document]
Comment (liamwhite) [reply src/video_core/rasterizer_interface.h:144]: Those break out of loops. This doesn't break out of anything.

[the remark this answers is not in the document]
Comment (GermanAizek) [reply src/video_core/rasterizer_interface.h:144]: @liamwhite ok.

[the remark this answers is not in the document]
Comment (GermanAizek) [reply src/video_core/rasterizer_interface.h]: @german77 fix it.
https://github.com/yuzu-emu/yuzu/pull/8268/commits/a49f1f57658f2ca294a65ea9581c5a9e4a72832f

File: src/yuzu/applets/qt_error.cpp [hunk 6]
```diff
@@ -24,7 +24,7 @@ void QtErrorDisplay::ShowError(ResultCode error, std::function<void()> finished)
         tr("An error has occurred.\nPlease try again or contact the developer of the software."));
 }
 
-void QtErrorDisplay::ShowErrorWithTimestamp(ResultCode error, std::chrono::seconds time,
+void QtErrorDisplay::ShowErrorWithTimestamp(const ResultCode& error, const std::chrono::seconds& time,
```
[the remark this answers is not in the document]
Comment (GermanAizek) [reply src/yuzu/applets/qt_error.cpp]: @merryhime fix it.
https://github.com/yuzu-emu/yuzu/pull/8268/commits/a49f1f57658f2ca294a65ea9581c5a9e4a72832f

File: src/video_core/vulkan_common/vulkan_wrapper.h [hunk 7]
```diff
@@ -625,7 +625,7 @@ class Queue {
     constexpr Queue(VkQueue queue_, const DeviceDispatch& dld_) noexcept
         : queue{queue_}, dld{&dld_} {}
 
-    VkResult Submit(Span<VkSubmitInfo> submit_infos,
+    VkResult Submit(const Span<VkSubmitInfo>& submit_infos,
```
[the remark this answers is not in the document]
Comment (GermanAizek) [reply src/video_core/vulkan_common/vulkan_wrapper.h]: @merryhime fix it.
https://github.com/yuzu-emu/yuzu/pull/8268/commits/a49f1f57658f2ca294a65ea9581c5a9e4a72832f

File: src/core/hle/service/ssl/ssl.cpp (hunk 4 above)
Comment (GermanAizek) [reply src/core/hle/service/ssl/ssl.cpp]: @merryhime fix it.
https://github.com/yuzu-emu/yuzu/pull/8268/commits/a49f1f57658f2ca294a65ea9581c5a9e4a72832f

File: src/core/frontend/applets/error.cpp (hunk 2 above)
Comment (GermanAizek) [reply src/core/frontend/applets/error.cpp]: @merryhime fix it.
https://github.com/yuzu-emu/yuzu/pull/8268/commits/a49f1f57658f2ca294a65ea9581c5a9e4a72832f

File: src/core/hle/service/es/es.cpp [hunk 8]
```diff
@@ -122,7 +122,6 @@ class ETicket final : public ServiceFramework<ETicket> {
 
     void ImportTicket(Kernel::HLERequestContext& ctx) {
         const auto ticket = ctx.ReadBuffer();
-        const auto cert = ctx.ReadBuffer(1);
```
[the remark this answers is not in the document]
Comment (GermanAizek) [reply src/core/hle/service/es/es.cpp]: @merryhime, @german77 fix it.
https://github.com/yuzu-emu/yuzu/pull/8268/commits/0ca29f1e546d630f65259ae7476ff44325b1b698

**Review by GermanAizek — COMMENTED** [review]
All errors and ideas were taken into account and eliminated, I expect second code review.

**Review by merryhime — CHANGES_REQUESTED** [review]
Incomplete review.

The reason this annoys me is because this feels like a few transformations a code linter could suggest, but I do not feel that the output of this linter has been reviewed by a human.

Therefore, I feel this PR is low value, given the effort one puts into reviewing it is essentially is the effort requried to write the PR properly in the first place.

I will not be reviewing further revisions of this PR and will leave it to yuzu's regular mantainers.

**Comment by Morph1984** [comment]
As @merryhime mentioned, a lot of these changes seem to be taken from a static analysis tool, and that the burden is on the maintainers / whoever is reviewing this code to ensure correctness.
Furthermore a number of these changes are done automatically without giving much thought into whether these are pessimizing changes (and thus further burdening the reviewer to ensure correctness).
Moreover, there are further changes that move the initialization of automatic variables to reduce their scope. These are intentionally done for: 1. These variables may be used in the future (not inside a branch), 2. Stylistic reasons, 3. Compilers can just reorder these variables in their optimization pass.

For these reasons and those that merry and others have mentioned, I'm afraid I will have to reject this PR.

**Pull request closed without merging**